lib/executor/ramping_arrival_rate.go

```diff
@@ -189,21 +188,11 @@ func (varr *RampingArrivalRate) Init(ctx context.Context) error {
 	et, err := varr.BaseExecutor.executionState.ExecutionTuple.GetNewExecutionTupleFromValue(varr.config.MaxVUs.Int64)
 	varr.et = et
 	start, offsets, lcd := et.GetStripedOffsets()
-	varr.segIdx = lib.NewSegmentedIndex(start, lcd, offsets)
+	varr.iterationSegIndex = lib.NewSegmentedIndex(start, lcd, offsets)
```

Comment:
Hhmm do we still need to redefine it here? Couldn't we use the instance created in `NewBaseExecutor()`, in which case we can get rid of this `Init()` entirely?

Reply:
I do think it is needed as this is based on the execution tuple created from the MaxVUs not just based on the execution segment alone.

If I remember correctly this meant that for some corner cases there will be ... wrong values returned otherwise:
execution segment 0:1/2,1/2:1 with 3 VUs will mean that 1 of the instances will have 2 VUs and there other 1 VU so if we are doing arrival rate of `3` the iterations should be 
```
1st instance: 1, 3, 4,6, ... 
2nd instance: 2,5,8 ... 
```
but if you just base it on the original execution segments they will be 
```
1st instance: 1,3,5,7 ...
2nd instance: 2,4,6,8 ...
```
It will still do the same amount of iteration just in the case that there needed to be 12(divides by 2 and 3 ;) ) iterations at some point there will be ... holes:
```
1st instance (has to do 2/3 of 12 so 8): 1,3,5,7,9,11,13,15
2nd instance (has to do 1/3 of 12 so 4):,2,4,6,8
```

P.s. I wrote this quickly so ... likely some of my math is slightly wrong and I also index from 1 ;)

Reply:
Yeah, because of differences like that, I'd prefer if we don't have `NextIterationCounters()` in `BaseExecutor` and instead have it as a helper. Maybe something like `GetISegmentedCounterIterator(start, lcd, offsets)` that returns essentially the `next()` closure?

Reply:
I think we should probably have 2 different types - a simple `SegmentedIndex` without any locking or synchronization, and a `SynchronizedSegmentedIndex` (or some better name...) that has locking and atomic operations.

Reply:
I don't think a second type is needed, as we have 1 place we use it and it's only for 1 method it is literally going to be 
```
func nextIteartionsCounters() (uint64, uint64){
   someMutex.Lock()
   defer someMutex.Unlock()
   segmentedIndex.next()
   return segmentedIndex.scaled, segmentedIndex.unscaled
}
```

I see no reason to make a whole new type for that .. .but it definitely is better so I am fine with that as well :)

Reply:
btw I had this pending as a comment in the main PR, but probably worth mentioning here: `SegmentedIndexResult` seems unnecessary. We are never going to have more than the scaled and unscaled indexes in this type, so a tuple (i.e. 2 return values) seems a bit more idiomatic in Go. 

If we ever add more things on top of `SegmentedIndex`, we are likely to do it as another wrapper, not by changing the original (synchronized or unsynchromized), so we can add a struct then, if we need to.